tests/vmi_networking_test.go

```diff
@@ -93,279 +92,264 @@ var _ = Describe("Networking", func() {
+		var cmdCheck, addrShow, addr string
+
+		// assuming pod network is of standard MTU = 1500 (minus 50 bytes for vxlan overhead)
+		expectedMtu := 1450
+		ipHeaderSize := 28 // IPv4 specific
+		payloadSize := expectedMtu - ipHeaderSize
+
+		// Wait until the VirtualMachineInstance is booted, ping google and check if we can reach the internet
+		switch destination {
+		case "Internet":
+			addr = "kubevirt.io"
+		case "InboundVMI":
+			addr = inboundVMI.Status.Interfaces[0].IP
+		case "InboundVMIWithPodNetworkSet":
+			addr = inboundVMIWithPodNetworkSet.Status.Interfaces[0].IP
 		}
+		fmt.Println(addr)
+
+		By("checking br1 MTU inside the pod")
+		vmiPod := tests.GetRunningPodByLabel(outboundVMI.Name, v1.DomainLabel, tests.NamespaceTestDefault)
+		output, err := tests.ExecuteCommandOnPod(
+			virtClient,
+			vmiPod,
+			vmiPod.Spec.Containers[0].Name,
+			[]string{"ip", "address", "show", "br1"},
+		)
+		log.Log.Infof("%v", output)
+		Expect(err).ToNot(HaveOccurred())
+		// the following substring is part of 'ip address show' output
+		expectedMtuString := fmt.Sprintf("mtu %d", expectedMtu)
+		Expect(strings.Contains(output, expectedMtuString)).To(BeTrue())
 
-		inboundVMI, outboundVMI = prepareVMIs(false)
-		// same but with explicit pod network definition
-		inboundVMI_podnet, outboundVMI_podnet = prepareVMIs(true)
-	})
+		By("checking eth0 MTU inside the VirtualMachineInstance")
+		expecter, err := tests.LoggedInCirrosExpecter(outboundVMI)
+		Expect(err).ToNot(HaveOccurred())
+		defer expecter.Close()
+
+		addrShow = "ip address show eth0\n"
+		resp, err := expecter.ExpectBatch([]expect.Batcher{
+			&expect.BSnd{S: "\n"},
+			&expect.BExp{R: "\\$ "},
+			&expect.BSnd{S: addrShow},
+			&expect.BExp{R: fmt.Sprintf(".*%s.*\n", expectedMtuString)},
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: "0"},
+		}, 180*time.Second)
+		log.Log.Infof("%v", resp)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("checking the VirtualMachineInstance can send MTU sized frames to another VirtualMachineInstance")
+		// NOTE: VirtualMachineInstance is not directly accessible from inside the pod because
+		// we transferred its IP address under DHCP server control, so the
+		// only thing we can validate is connectivity between VMIs
+		//
+		// NOTE: cirros ping doesn't support -M do that could be used to
+		// validate end-to-end connectivity with Don't Fragment flag set
+		cmdCheck = fmt.Sprintf("ping %s -c 1 -w 5 -s %d\n", addr, payloadSize)
+		err = tests.CheckForTextExpecter(outboundVMI, []expect.Batcher{
+			&expect.BSnd{S: "\n"},
+			&expect.BExp{R: "\\$ "},
+			&expect.BSnd{S: cmdCheck},
+			&expect.BExp{R: "\\$ "},
+			&expect.BSnd{S: "echo $?\n"},
+			&expect.BExp{R: "0"},
+		}, 180)
+		Expect(err).ToNot(HaveOccurred())
+	},
+		table.Entry("the Inbound VirtualMachineInstance", "InboundVMI"),
+		table.Entry("the Inbound VirtualMachineInstance with pod network connectivity explicitly set", "InboundVMIWithPodNetworkSet"),
```

Comment:
Most changes are just indenting: this is important.

Reply:
@booxter, here I added an extra entry. This now checks if a vm which explicitly requested the pod network is still accessible. We could also add one regarding to outbound connections.